### PR TITLE
QueryTypes: Add feature toggle to show query types in datasource apiservers

### PR DESCRIPTION
**Pull request by ryantxu**
This PR adds a new feature toggle `datasourceQueryTypes` that will add the querytypes resource to apiserves that can show them:
* http://localhost:3000/apis/query.grafana.app/v0alpha1/querytypes
* http://localhost:3000/apis/testdata.datasource.grafana.app/v0alpha1/querytypes
* http://localhost:3000/apis/prometheus.datasource.grafana.app/v0alpha1/querytypes

These each have a "validate" stub where we can explore hooking up [real validation/mutation hooks ](https://github.com/grafana/grafana/pull/87718)

### CODE REVIEW

File: pkg/registry/apis/query/queryschema/oas_helper.go [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+	PluginJSON *plugins.JSONData
+	QueryTypes *query.QueryTypeDefinitionList
+
+	Root             string
+	QueryPath        string // eg "namespaces/{namespace}/query/{name}"
+	QueryDescription string
+	QueryExamples    map[string]*spec3.Example
+}
+
+func AddQueriesToOpenAPI(options OASQueryOptions) error {
```
Comment (ryantxu) [inline pkg/registry/apis/query/queryschema/oas_helper.go:34]: This moves the logic that was duplicated in the query and datasource apiservers -- it now produces a more valid response (still a few issues in converting jsonschema to OpenAPI, but at least it is in one place now!)

**Review by andresmgot — COMMENTED** [review]
I think this makes sense. What would be the plan for external plugins?

**Comment by ryantxu** [comment]
for external (and internal!) we need to decide how the plugin will advertise these additional settings.  I think the simplest solution is to publish static files along with the plugin and load them into something like the pfs.

This is likely where we will also want to stick the apiversion(s) and other information we may want for the apiserver.

But that is for another day :)

**Review by andresmgot — APPROVED** [review]
Sounds good to me, I guess we need more eyes here

**Pull request merged**